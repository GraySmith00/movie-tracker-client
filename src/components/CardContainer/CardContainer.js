import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const CardContainer = ({ movies, category }) => {
  const displayCards = movies[category].map(movie => (
    <MovieCard key={movie.id} movie={movie} />
  ));

  return <div>{displayCards}</div>;
};

CardContainer.propTypes = {
  movies: PropTypes.object.isRequired,
  category: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(mapStateToProps)(CardContainer);
