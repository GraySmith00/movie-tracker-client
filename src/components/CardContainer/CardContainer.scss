.card-container {
  margin: 2rem auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);

  grid-gap: 2em;
  width: 90%;
}
