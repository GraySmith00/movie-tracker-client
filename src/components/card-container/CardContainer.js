import React from "react";
import MovieCard from "../movie-card/MovieCard";

const CardContainer = ({ data, path }) => {
  const cardContainer = data.map((movieObj, index) => (
    <MovieCard key={`Container-${index}`} movieObj={movieObj} path={path} />
  ));
  return <div> {cardContainer}</div>;
};

export default CardContainer;
