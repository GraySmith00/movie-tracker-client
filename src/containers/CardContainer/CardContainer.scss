.card-container {
  margin: 2rem auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);

  grid-gap: 2em;
  width: 90%;
}

@media screen and (max-width: 1340px) {
  .card-container {
    grid-gap: 0.8em;
  }
}

@media screen and (max-width: 1280px) {
  .card-container {
    grid-gap: 2em;
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 1060px) {
  .card-container {
    grid-gap: 0.8em;
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (max-width: 1000px) {
  .card-container {
    grid-gap: 2em;
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (max-width: 540px) {
  .card-container {
    grid-template-columns: repeat(1, 1fr);
    justify-content: center;
    padding: 0 25%;
  }
}
