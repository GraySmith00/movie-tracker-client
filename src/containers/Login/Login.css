.login-user-form {
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 2.5rem 0 0 0; }

.login-email {
  height: 3rem;
  width: 25rem;
  font-size: 1rem;
  margin-bottom: 1.5rem;
  border-radius: 0.5rem; }

.login-password {
  height: 3rem;
  width: 25rem;
  font-size: 1rem;
  margin-bottom: 1.5rem;
  border-radius: 0.5rem; }

.login-submit-button {
  background-color: #00aad5;
  height: 3rem;
  width: 25rem;
  font-size: 1.5rem;
  border-radius: 0.5rem;
  color: #fff; }

.error-message {
  text-align: center;
  color: crimson; }
