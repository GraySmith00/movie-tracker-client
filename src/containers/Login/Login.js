import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser, getFavorites } from '../../helpers';
import { setCurrentUser } from '../../actions/userActions';
import { populateFavoritesState } from '../../actions/movieActions';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  setFavoritesState = async () => {
    const { currentUser, populateFavoritesState } = this.props;
    if (currentUser) {
      const favorites = await getFavorites(currentUser);
      const movieIds = favorites.data.map(favorite => favorite.movie_id);
      populateFavoritesState(movieIds);
    }
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { email, password } = this.state;
    try {
      const currentUser = await loginUser(email, password);
      this.props.setCurrentUser(currentUser);
      this.setState({
        email: '',
        password: ''
      });

      if (currentUser) {
        this.setFavoritesState();
        this.props.history.push('/');
      }
    } catch (error) {
      alert(error.message);
    }
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          onChange={this.handleChange}
          type="text"
          name="email"
          value={this.state.email}
          placeholder="Email..."
        />
        <input
          onChange={this.handleChange}
          type="text"
          name="password"
          value={this.state.password}
          placeholder="Password..."
        />
        <button>Submit</button>
      </form>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user)),
  populateFavoritesState: movieIds => dispatch(populateFavoritesState(movieIds))
});

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
