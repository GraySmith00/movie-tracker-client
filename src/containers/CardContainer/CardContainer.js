import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './CardContainer.css';

export const CardContainer = ({ movies, category }) => {
  let displayCards;
  if (category === 'favorites') {
    if (movies.favorites.length === 0) {
      displayCards = <p>You have not added anything to your favorites yet!</p>;
    } else {
      displayCards = movies[category].map(movieId => {
        const movie = movies.nowPlaying.find(
          movie => movie.movie_id === movieId
        );
        return <MovieCard key={movie.movie_id} movie={movie} />;
      });
    }
  } else {
    displayCards = movies[category].map(movie => {
      return <MovieCard key={movie.movie_id} movie={movie} />;
    });
  }

  return <div className="card-container">{displayCards}</div>;
};

CardContainer.propTypes = {
  movies: PropTypes.object.isRequired,
  category: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(mapStateToProps)(CardContainer);
