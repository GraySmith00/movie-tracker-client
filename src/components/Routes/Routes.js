import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from '../../containers/Login/Login';
import Register from '../../containers/Register/Register';
import CardContainer from '../../containers/CardContainer/CardContainer.js';

export const Routes = () => {
  return (
    <Switch>
      <Route
        exact
        path="/"
        render={() => {
          return <CardContainer category={'nowPlaying'} />;
        }}
      />
      <Route exact path="/login" component={Login} />
      <Route exact path="/register" component={Register} />
      <Route
        exact
        path="/favorites"
        render={() => {
          return <CardContainer category={'favorites'} />;
        }}
      />
    </Switch>
  );
};

export default Routes;
