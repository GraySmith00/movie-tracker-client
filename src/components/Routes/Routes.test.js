import React from 'react';
import { shallow } from 'enzyme';
import { Routes } from './Routes';
import Login from '../../containers/Login/Login';
import Register from '../../containers/Register/Register';
import { CardContainer } from '../../containers/CardContainer/CardContainer';

describe('Routes', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Routes />);
  });

  it('should match snapshot of nowPlaying Route', () => {
    const category = 'nowPlaying';
    wrapper = shallow(
      <Routes
        path="/"
        render={() => {
          return <CardContainer category={category} />;
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapShot of login Route', () => {
    let loginWrapper = shallow(<Routes path="/login" component={Login} />);
    expect(loginWrapper).toMatchSnapshot();
  });

  it('should match snapShot of register Route', () => {
    let registerWrapper = shallow(
      <Routes path="/register" component={Register} />
    );
    expect(registerWrapper).toMatchSnapshot();
  });
});
