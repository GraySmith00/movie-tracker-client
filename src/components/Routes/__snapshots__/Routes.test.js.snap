// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should match snapShot of login Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes
    component={[Function]}
    path="/login"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/register"
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/register"
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Routes should match snapShot of register Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes
    component={[Function]}
    path="/register"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/register"
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/register"
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Routes should match snapshot of favorites Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    initialIndex={
        Array [
            Object {
              "key": 0,
              "pathname": "/",
            },
          ]
    }
>
    <Routes
        path="/favorites"
        render={[Function]}
    />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Routes
        path="/favorites"
        render={[Function]}
/>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "x5iwga",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": NaN,
        "length": 1,
        "listen": [Function],
        "location": undefined,
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "path": "/favorites",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Routes
          path="/favorites"
          render={[Function]}
/>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "x5iwga",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": NaN,
          "length": 1,
          "listen": [Function],
          "location": undefined,
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Routes should match snapshot of nowPlaying Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes
    path="/"
    render={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/register"
/>,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/register"
/>,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
